project-management-openproject/project-management-openproject-api/src/main/java/com/xwiki/projectmanagement/openproject/internal/processing/OpenProjectFilterHandler.java

```diff
@@ -50,89 +52,195 @@ private OpenProjectFilterHandler()
 
         for (LiveDataQuery.Filter filter : filters) {
 
-            List<LiveDataQuery.Constraint> validConstraints = filter
-                .getConstraints()
-                .stream()
-                .filter(constraint -> {
-                    String constraintValue = (String) constraint.getValue();
-                    return constraintValue != null && !constraintValue.isEmpty();
-                }).collect(Collectors.toList());
+            List<LiveDataQuery.Constraint> validConstraints = getValidConstraints(filter);
 
             if (validConstraints.isEmpty()) {
                 continue;
             }
-            List<Map<String, Object>> filterConstraints = new ArrayList<>();
-            for (LiveDataQuery.Constraint constraint : validConstraints) {
-                String operatorValue = OpenProjectMapper.mapLivedataOperator(constraint.getOperator());
-                Map<String, Object> existingOperator =
-                    filterConstraints.stream()
-                        .filter(fc -> fc.get(OPERATOR).equals(operatorValue))
-                        .findFirst().orElse(null);
-
-                if (existingOperator == null) {
-                    Map<String, Object> newOperator = new HashMap<>();
-                    newOperator.put(OPERATOR, operatorValue);
-                    List<String> initialValues = new ArrayList<>();
-                    initialValues.add((String) constraint.getValue());
-                    newOperator.put(VALUES, initialValues);
-                    filterConstraints.add(newOperator);
-                } else {
-                    List<String> currentValues = (List<String>) existingOperator.get(VALUES);
-                    currentValues.add((String) constraint.getValue());
-                    existingOperator.put(VALUES, currentValues);
-                }
-            }
 
-            for (Map<String, Object> filterConstraint : filterConstraints) {
-                Map<String, Object> convertedFilter = new HashMap<>();
-                convertedFilter.put(OpenProjectMapper.mapLivedataProperty(filter.getProperty()), filterConstraint);
-                convertedFilters.add(convertedFilter);
-            }
+            Map<String, Object> filterConstraints = handleFilterConstraints(validConstraints);
+            String filterName = OpenProjectMapper.mapLivedataProperty(filter.getProperty());
+            mergedFilters.put(filterName, filterConstraints);
         }
 
-        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> convertedFiltersFromString = handleFilterStringConstraints(filtersString);
+
+        mergeFiltersMaps(mergedFilters, convertedFiltersFromString);
+
+        return convertMapOfFiltersToString(mergedFilters);
+    }
+
+    private static Map<String, Object> handleFilterStringConstraints(String filtersString)
+        throws ProjectManagementException
+    {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
I see you use multiple object mappers. Turns out they are thread safe. You can declare a static one and use that.